gatsby/config/gatsby-config.ts

```diff
@@ -1,3 +1,5 @@
+import dotenv from 'dotenv';
+dotenv.config({ path: `.env.${process.env.NODE_ENV}` });
```

Comment:
to  funjuje automaticky bez `dotenv` https://www.gatsbyjs.com/docs/environment-variables/#client-side-javascript